src/main/java/fruit/shop/service/impl/StringConnectorImpl.java

```diff
@@ -0,0 +1,19 @@
+package fruit.shop.service.impl;
+
+import fruit.shop.service.StringConnector;
+import java.util.Map;
+
+public class StringConnectorImpl implements StringConnector {

```

Comment:
You should use database
All magic numbers, strings, symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)
Code should be similar like this:
```
public class ReportServiceImpl implements ReportService {
    private static final String TITLE = "fruit,quantity\n";
    private static final String SEPARATOR = ",";

    @Override
    public String generateReport() {
        String report = TITLE;
        report += Storage.FRUITS.keySet().stream()
                .map(fruit -> fruit + SEPARATOR + STORAGE.get(fruit) + System.lineSeparator())
                .collect(Collectors.joining());
        return report;
    }
}
```

Reply:
not fixed

Reply:
What exactly is not fixed?

Reply:
Do you mean implementation using Stream API?
I can just copy paste, it's not problem. But why my way is incorrect?
There is no specific task how to implement it, so I did it how I see it.